pyro/poutine/indep_messenger.py

```diff
@@ -108,7 +109,7 @@ def _reset(self):
     @property
     def indices(self):
         if self._indices is None:
-            self._indices = torch.arange(self.size, dtype=torch.long)
+            self._indices = torch.arange(self.size, dtype=torch.long).to(self.device)
```

Comment:
just curious, why `.to(self.device)` rather than `dtype=torch.long, device=self.device`? Is `arange` not implemented for cuda in some versions of PyTorch?

Reply:
I just copied this from `_Subsample`, no deeper reason